client/src/main/scala/org/apache/celeborn/client/LifecycleManager.scala

```diff
@@ -79,6 +81,13 @@ class LifecycleManager(val appUniqueId: String, val conf: CelebornConf) extends
     JavaUtils.newConcurrentHashMap[Int, ConcurrentHashMap[Int, PartitionLocation]]()
   private val userIdentifier: UserIdentifier = IdentityProvider.instantiate(conf).provide()
   private val availableStorageTypes = conf.availableStorageTypes
+  // app shuffle id -> LinkedHashMap of (app shuffle identifier, (shuffle id, fetch status))
+  private val shuffleIdMapping = JavaUtils.newConcurrentHashMap[
+    Int,
+    scala.collection.mutable.LinkedHashMap[String, (Int, Boolean)]]()
+  private val shuffleIdGenerator = new AtomicInteger(0)
+  // app shuffle id -> whether shuffle is determinate, rerun of a indeterminate shuffle gets different result
+  private val appShuffleDeterminateMap = JavaUtils.newConcurrentHashMap[Int, Boolean]();
```

Comment:
Do we need to maintain this state? Can the `GetShuffleId` message not include whether the shuffle is determinate or not?

Reply:
I like the idea of reducing the status~
however, it seems hard to get rdd/stage DeterministicLevel in task/executor, it is where GetShuffleId RPC is called